src/input.rs

```diff
@@ -757,37 +663,163 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
         }
     }
 
+    fn update_click_state(&mut self, button: MouseButton) {
+        let elapsed = self.ctx.mouse().last_click_timestamp.elapsed();
+        let button_changed = self.ctx.mouse().last_button != button;
+        let click_state = self.ctx.mouse().click_state;
+
+        self.ctx.mouse_mut().last_click_state = click_state;
```

Comment:
I just realised some of my code changes got lost in a rebase. I'm not using `last_click_state` anywhere, which was added to mimic the exact workings of the previous `on_mouse_press` `match` behaviour.

You should probably hold off your review until I manage to put the missing code back in place.

Reply:
Thanks for letting me know. Please ping me when you think that everything's ready to go and I'll get on it asap.